heat/core/dndarray.py

```diff
@@ -62,6 +64,9 @@ def __init__(self, array, gshape, dtype, split, device, comm):
         ):
             self.__array = self.__array.to(devices.sanitize_device(self.__device).torch_device)
 
+        if self.__halo > 0:
```

Comment:
Why would axis zero not have halos? Is '0' used as placeholder for no split? If so, 'None' seems more meaningful as default.

Reply:
self.__halo is the with of the halo not the axis. The axis is always the same as self.split